src/__tests__/JWE.test.ts

```diff
@@ -186,4 +215,152 @@ describe('JWE', () => {
       })
     })
   })
-})
+
+  describe('ECDH-1PU+XC20PKW (X25519), Key Wrapping Mode with XC20P content encryption', () => {
+    describe('One recipient', () => {
+      let cleartext, recipientKey, senderKey, decrypter
+
+      beforeEach(() => {
+        recipientKey = generateKeyPairFromSeed(randomBytes(32))
+        senderKey = generateKeyPairFromSeed(randomBytes(32))
+        cleartext = u8a.fromString('my secret message')
+        decrypter = x25519AuthDecrypter(recipientKey.secretKey, senderKey.publicKey)
+      })
+
+      it('Creates with only ciphertext', async () => {
+        const encrypter = x25519AuthEncrypter(recipientKey.publicKey, senderKey.secretKey)
+        expect.assertions(3)
+        const jwe = await createJWE(cleartext, [encrypter])
+        expect(jwe.aad).toBeUndefined()
+        expect(JSON.parse(decodeBase64url(jwe.protected))).toEqual({ enc: 'XC20P' })
+        expect(await decryptJWE(jwe, decrypter)).toEqual(cleartext)
+      })
+
+      it('Creates with skid, kid, no apu and no apv', async () => {
+        const kid = 'did:example:receiver#key-1'
+        const skid = 'did:example:sender#key-1'
+        const encrypter = x25519AuthEncrypter(recipientKey.publicKey, senderKey.secretKey, { kid, skid } )
+        expect.assertions(6)
+        const jwe = await createJWE(cleartext, [encrypter])
+        expect(jwe.aad).toBeUndefined()
+        expect(JSON.parse(decodeBase64url(jwe.protected))).toEqual({ enc: 'XC20P', skid: skid })
+        expect(jwe.recipients[0].header.kid).toEqual(kid)
+        expect(jwe.recipients[0].header.apu).toEqual(encodeBase64url(skid))
```

Comment:
How would this work with an encrypted skid?

Reply:
`skid` as per ECDH-1PU is just an OPTIONAL `kid` of the sender in the protected header. The discussion in the DIDComm WG will likely conclude with either including a `encrypted_skid` (or `encrypted_spk`) in the recipient header. So, this is unrelated. In this implementation `skid` is just a blob without any meaning similar to `kid`.

Adding `encrypted_skid` (or `encrypted_spk`) could be subject to another PR. Would you agree?

Reply:
Sure, was just thinking that passing an object here (JWE) rather than a string will likely break the current implementation.